lib/fetch/util.js

```diff
@@ -286,7 +286,160 @@ function clonePolicyContainer () {
 // https://w3c.github.io/webappsec-referrer-policy/#determine-requests-referrer
 function determineRequestsReferrer (request) {
   // TODO
-  return 'no-referrer'
+  // 1. Let policy be request's referrer policy.
+  const policy = request.referrerPolicy
+
+  // Return no-referrer when empty or policy says so
+  if (policy === '' || policy === 'no-referrer') return 'no-referrer'
+
+  // 2. Let environment be the request client
+  const environment = request.client
+  let referrerSource = null
+
+  // TODO
+  // 3(a) If policy is "origin", then
+  // 3(a)(I) If environment's global object is Window then
+  if (request.referrer === 'client') {
```

Comment:
Why this? Not in spec?

Reply:
Accordingly to the third step of the spec, states a switch on the `referrer` value, but wasn’t sure if we want it to totally cover it or now, so I just applied as spec stated:

>3, Switch on request’s [referrer](https://fetch.spec.whatwg.org/#concept-request-referrer):
- "client"
  - If environment’s [global object](https://html.spec.whatwg.org/multipage/webappapis.html#concept-settings-object-global) is a [Window](https://html.spec.whatwg.org/multipage/window-object.html#window) object, then
     - Let document be the [associated Document](https://html.spec.whatwg.org/multipage/browsers.html#concept-document-window) of environment’s [global object](https://html.spec.whatwg.org/multipage/webappapis.html#concept-settings-object-global).
       - If document’s [origin](https://dom.spec.whatwg.org/#concept-document-origin) is an [opaque origin](https://html.spec.whatwg.org/multipage/browsers.html#concept-origin-opaque), return no referrer.
       - While document is [an iframe srcdoc document](https://html.spec.whatwg.org/multipage/embedded-content.html#an-iframe-srcdoc-document), let document be document’s [browsing context](https://html.spec.whatwg.org/multipage/browsers.html#concept-document-bc)’s [browsing context container](https://html.spec.whatwg.org/multipage/browsers.html#browsing-context-container)’s [node document](https://dom.spec.whatwg.org/#concept-node-document).
      - Let referrerSource be document’s [URL](https://dom.spec.whatwg.org/#concept-document-url).
  - Otherwise, let referrerSource be environment’s [creation URL](https://html.spec.whatwg.org/multipage/webappapis.html#concept-environment-creation-url).
a [URL](https://url.spec.whatwg.org/#concept-url)
- Let [referrer](https://fetch.spec.whatwg.org/#concept-request-referrer)Source be request’s referrer.

Reply:
Look at how it's done in the rest of the existing code.

Reply:
Sure thing, I'll do that 👍